fdbserver/VersionedBTree.actor.cpp

```diff
@@ -5680,7 +5644,9 @@ class VersionedBTree {
 					domainId = currentDomainId;
 					domainPrefixLength = prefixLength;
 					canUseDefaultDomain =
-					    (height > 1 && (currentDomainId == defaultDomainId || prefixLength == rec.key.size()));
+					    (height > 1 && (currentDomainId == defaultDomainId ||
+					                    (prefixLength == rec.key.size() &&
```

Comment:
Why is this ==?  Keys that are shorter than prefixLength can also be encrypted with the default domain, and it seems like that would also occur at the top of the tree.

Reply:
For any key shorter than the prefix, the key provider would return default domain id as its domain id, so it is already covered by the `currentDomainId == defaultDomainId` condition.